app/grandchallenge/workstations/static/workstations/js/session-control.mjs

```diff
@@ -101,6 +103,24 @@ function copyTextToClipboard(text) {
     });
 }
 
+function setSpinner(element) {
+    element.querySelector("i").style.display = "none";
+    const spinner = document.createElement("span");
+    spinner.classList.add("spinner-border", "spinner-border-sm");
+    element.prepend(spinner);
+}
+
+function removeSpinner(element) {
+    const spinner = element.querySelector("span");
```

Comment:
Would the cleaner way, rather than selecting on `<span>`, be to select on `spinner-border` class? I could forsee someone adding a <span> within a button.